test/e2e/tests/validator_test.go

```diff
@@ -178,10 +178,11 @@ func (s *validatorSchedule) Increment(heights int64) {
 	}
 }
 
-func makeVals(valMap map[*e2e.Node]int64) []*types.Validator {
+func makeVals(testnet *e2e.Testnet, valMap map[string]int64) []*types.Validator {
 	vals := make([]*types.Validator, 0, len(valMap))
-	for node, power := range valMap {
-		vals = append(vals, types.NewValidator(node.PrivvalKey.PubKey(), power))
+	for valName, power := range valMap {
+		validator := testnet.LookupNode(valName)
+		vals = append(vals, types.NewValidator(validator.PrivvalKey.PubKey(), power))
```

Comment:
the `LookupNode` can return `nil`, should test for that and only append in that case ? not sure if there are corner cases that can make it return `nil` but just highlighting the fact here, but it would panic if `validator` is `nil` too

Reply:
This applies to other parts of the refactorings above as well.

Reply:
Good point folks. I'll revisit the PR with this in mind

Reply:
Addressed